### PR TITLE
OS: Added CoreMidi Support

### PR DESCRIPTION
**Pull request by CodeforEvolution**
Hello! This currently builds the qtplaysmf demo, though I'm otherwise unable to test this for now with my Mac. Besides only supporting normal midi messages for
QMidiIn, this should add full support for CoreMIDI in QMidi.

### CODE REVIEW

File: src/OS/QMidi_CoreMidi.cpp [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+	MIDISendSysex(&request);
+}
+
+// # pragma mark - QMidiIn
+
+struct NativeMidiInInstances {
+  MIDIClientRef client;
+  MIDIPortRef inputPort;
+  MIDIEndpointRef sourceId;
```
Comment (waddlesplash) [inline src/OS/QMidi_CoreMidi.cpp]: Indentation.

**Comment by waddlesplash** [comment]
At last! This has been on the wishlist since 2013 when I first posted this library. Thanks!

**Comment by CodeforEvolution** [comment]
No problem! I wanted to learn a bit more about MIDI anyway! :D

**Comment by CodeforEvolution** [comment]
Hey @waddlesplash! While this should be functional now, sadly, some APIs in CoreMIDI that I was using for MIDI handling are going to be deprecated in the next version of Apple’s operating systems (Revealed today actually, the irony :D). Should I try to prepare for these new APIs? (The APIs are still in beta however)

**Comment by waddlesplash** [comment]
Well, for now, let's leave this as-is.

**Pull request merged**